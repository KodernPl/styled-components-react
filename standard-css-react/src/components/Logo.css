@import '../styles/variables.css';

.logo {
  position: absolute;
  top: 5rem;
  left: 30%;
  width: 40%;
  animation: logoAnimation 4s linear infinite;

  &--light {
    background: var(--gradient-light);
  }

  &--dark {
    background: var(--gradient-dark);
  }
}

@keyframes logoAnimation {
  0% {
    transform: rotate(0deg);
  }

  25% {
    transform: rotate(5deg);
  }

  50% {
    transform: rotate(0deg);
  }

  75% {
    transform: rotate(-5deg);
  }
}
