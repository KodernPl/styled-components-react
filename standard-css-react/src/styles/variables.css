:root {
  --color-white: white;
  --color-gradient1: #667db6;
  --color-gradient2: #0082c8;
  --color-gradient3: #0082c8;
  --color-gradient4: #667db6;
  --color-white: #fff;
  --color-black: #000;
  --color-dark-grey: #222;
  --color-darkest-grey: #111;
  --color-grey: grey;
  --color-palevioletred: palevioletred;
  --color-sc-pink: rgb(219, 112, 147);
  --color-sc-orange: rgb(218, 163, 87);
  --color-alpha-black: rgba(0, 0, 0, 0.2);
  --color-alpha-grey: rgba(20, 20, 20, 0.1);
  --gradient-light: linear-gradient(to top right, rgb(219, 112, 147), rgb(218, 163, 87));
  --gradient-dark: linear-gradient(to top right, #667db6, #0082c8, #0082c8);
}

@custom-media --desktop (min-width: 1024px);
