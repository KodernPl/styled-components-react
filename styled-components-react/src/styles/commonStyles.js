import styled from 'styled-components';

export const CenterElements = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

export const CustomLink = styled.a`
  display: inline-block;
  width: 20rem;
  padding: .5rem 0;
  margin: .5rem 1rem;
  border-width: .2rem;
  border-radius: .3rem;
  border-style: solid;
  border-image: initial;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  font-size: 1.2rem;
`;
