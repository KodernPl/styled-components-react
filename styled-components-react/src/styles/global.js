import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
  html {
    box-sizing: border-box;
    font-size: 62.5%;
    font-family: sans-serif;
  }

  body {
    margin: 0;
    padding: 0;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }
`;
