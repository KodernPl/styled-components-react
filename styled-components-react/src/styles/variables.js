export const COLORS = {
  gradient1: '#667db6',
  gradient2: '#0082c8',
  gradient3: '#0082c8',
  gradient4: '#667db6',

  white: '#fff',
  black: '#000',
  darkGrey: '#222',
  darkestGrey: '#111',

  grey: 'grey',
  palevioletred: 'palevioletred',

  scPink: 'rgb(219, 112, 147)',
  scOrange: 'rgb(218, 163, 87)',

  alphaBlack: 'rgba(0, 0, 0, 0.2)',
  alphaGrey: 'rgba(20, 20, 20, 0.1)',
};

export const THEMES = {
  light: 'light',
  dark: 'dark',
};

export const GRADIENTS = {
  light: `linear-gradient(to top right, ${COLORS.scPink}, ${COLORS.scOrange})`,
  dark: `linear-gradient(to top right, ${COLORS.gradient1}, ${COLORS.gradient2}, ${COLORS.gradient4})`,
};

export const BREAKPOINTS = {
  desktop: '(min-width: 1024px)',
};
