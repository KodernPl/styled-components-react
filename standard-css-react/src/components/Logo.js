import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import { withTheme } from '../utils/themeContext';
import { THEMES } from '../styles/variables';

import './Logo.css';

const ns = 'logo';
const displayName = 'Logo';

const propTypes = {
  theme: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

const Logo = (props) => {
  const { theme, alt } = props;
  const classNames = cx(
    [`${ns}`],
    {
      [`${ns}--light`]: theme === THEMES.light,
      [`${ns}--dark`]: theme === THEMES.dark,
    },
  );
  return (
    <img className={classNames} alt={alt} {...props} />);
};

Logo.displayName = displayName;
Logo.propTypes = propTypes;

export default withTheme(Logo);
