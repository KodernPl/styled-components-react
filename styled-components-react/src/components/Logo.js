import React from 'react';
import styled, { keyframes } from 'styled-components';
import styledTheming from 'styled-theming';

import { GRADIENTS, THEMES } from '../styles/variables';

const gradient = styledTheming('mode', {
  [THEMES.light]: GRADIENTS.light,
  [THEMES.dark]: GRADIENTS.dark,
});

const logoAnimation = keyframes`
  0% {
    transform: rotate(0deg);
  }

  25% {
    transform: rotate(5deg);
  }

  50% {
    transform: rotate(0deg);
  }

  75% {
    transform: rotate(-5deg);
  }
`;

const StyledLogo = styled.img`
  position: absolute;
  top: 5rem;
  left: 30%;
  width: 40%;
  animation: ${logoAnimation} 4s linear infinite;
  background-image: ${gradient};
`;

const displayName = 'Logo';

const Logo = props => <StyledLogo {...props} />;

Logo.displayName = displayName;

export default Logo;
