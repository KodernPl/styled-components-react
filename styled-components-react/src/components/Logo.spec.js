import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import { THEMES } from '../styles/variables';

import Logo from './Logo';

const mockSrc = 'TEST_URL';
const mockAlt = 'TEST_IMAGE';

const mockComponent = theme => (
  <ThemeProvider theme={{ mode: theme }}>
    <Logo
      src={mockSrc}
      alt={mockAlt}
    />
  </ThemeProvider>
);
[
  THEMES.light,
  THEMES.dark,
].forEach((theme) => {
  describe(`${theme} theme`, () => {
    const cmp = renderer.create(mockComponent(theme)).toJSON();
    test('renders correct HTML', () => {
      expect(cmp).toMatchSnapshot();
    });

    describe('image props and styles', () => {
      test('should have correct values', () => {
        expect(cmp.props.src).toEqual(mockSrc);
        expect(cmp.props.alt).toEqual(mockAlt);
      });

      test('should have width 40%', () => {
        expect(cmp).toHaveStyleRule('width', '40%');
      });
    });
  });
});
