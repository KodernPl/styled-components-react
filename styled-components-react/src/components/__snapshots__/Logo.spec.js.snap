// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark theme renders correct HTML 1`] = `
.c0 {
  position: absolute;
  top: 5rem;
  left: 30%;
  width: 40%;
  -webkit-animation: drfsvC 4s linear infinite;
  animation: drfsvC 4s linear infinite;
  background-image: linear-gradient(to top right,#667db6,#0082c8,#667db6);
}

<img
  alt="TEST_IMAGE"
  className="c0"
  src="TEST_URL"
/>
`;

exports[`light theme renders correct HTML 1`] = `
.c0 {
  position: absolute;
  top: 5rem;
  left: 30%;
  width: 40%;
  -webkit-animation: drfsvC 4s linear infinite;
  animation: drfsvC 4s linear infinite;
  background-image: linear-gradient(to top right,rgb(219,112,147),rgb(218,163,87));
}

<img
  alt="TEST_IMAGE"
  className="c0"
  src="TEST_URL"
/>
`;
